memo_app.rb

```diff
@@ -2,20 +2,45 @@
 
 require 'sinatra'
 require 'sinatra/reloader'
-require 'json'
+require 'pg'
 
 helpers do
   def h(text)
     Rack::Utils.escape_html(text)
   end
 end
 
+def conn
+  @conn ||= PG.connect(dbname: 'memodb', client_encoding: 'UTF8')
+end
+
+before do
+  conn.exec('CREATE TABLE IF NOT EXISTS memos (id serial, title varchar(255), content text)')
+end
+
 def load_memos
-  File.exist?('memo.json') ? JSON.parse(File.read('memo.json')) : {}
+  result = conn.exec('SELECT * FROM memos')
+  memos = {}
+  result.each do |row|
+    memos[row['id']] = {
+      'id' => row['id'],
+      'title' => row['title'],
+      'content' => row['content']
+    }
+  end
+  memos
```

Comment:
豆知識的なものですが、 `each_with_object` を使うと少し簡単に書けます
```suggestion
  result.each_with_object({}) do |row, memos|
    memos[row['id']] = {
      'id' => row['id'],
      'title' => row['title'],
      'content' => row['content']
    }
  end
```
https://docs.ruby-lang.org/ja/latest/method/Enumerable/i/each_with_object.html

Reply:
こちらのやり方がシンプルなのでこちらに変更しました！